lib/ASTGen/Sources/ASTGen/Generics.swift

```diff
@@ -39,51 +39,37 @@ extension ASTGenVisitor {
     _ node: GenericParameterClauseSyntax,
     action: (BridgedArrayRef, BridgedArrayRef) -> T
   ) -> T {
-    var params = [UnsafeMutableRawPointer]()
```

Comment:
Not a fan of this method in general, I got rid of it down the line in the "translate `where` clauses" commit . That said, I agree the functional version is not as straightforward, but.. that bad? Could the cluttered diff have affected your impression?

> Do you have any numbers on what the performance gain here is?

The motivation for `map(_: withBridgedArrayRef:)` is as much fast path provision as encapsulation, but regardless, I’m a kind of person that doesn’t need a bad metric score to do a refactor of this sort and scope. This here is really just me being consistent across all occurrences of the `map` + `withBridgedArrayRef` pattern though.

Reply:
I just commented on the first one I saw here, but my comment was more generally about `map(_: withBridgedArrayRef:)` than this method itself. Personally I find this much harder to read because of the nesting, but if I'm the only one that thinks that... I don't have a better alternative 🤷